libs/core/kiln_ai/tools/kiln_task_tool.py

```diff
@@ -0,0 +1,160 @@
+from functools import cached_property
+from typing import Any, Dict
+
+from kiln_ai.datamodel import Task
+from kiln_ai.datamodel.external_tool_server import ExternalToolServer
+from kiln_ai.datamodel.task import TaskRunConfig
+from kiln_ai.datamodel.task_output import DataSource, DataSourceType
+from kiln_ai.datamodel.tool_id import ToolId
+from kiln_ai.tools.base_tool import KilnToolInterface, ToolCallContext
+from kiln_ai.utils.project_utils import project_from_id
+
+
+class KilnTaskToolResult:
```

Comment:
Check out  Python`@dataclass`. Let's you skip writing an init method. Nothing wrong with this, just good to know.